### PR TITLE
Add ability to log to auditor

### PR DESCRIPTION
**Pull request by Firesphere**
- Add auditor to suggested modules
- Log all steps to logger

Issue: https://github.com/Firesphere/silverstripe-bootstrapmfa/issues/33

### CODE REVIEW

**Comment by Firesphere** [comment]
Failures of tests are fixed in a separate PR

File: src/Handlers/BootstrapMFALoginHandler.php [hunk 1]
```diff
@@ -260,4 +272,37 @@ protected function cancelLogin(HTTPRequest $request)
...
+     * @param int $type
+     */
+    protected function log($message, $type)
+    {
+        $memberID = $this->getRequest()->getSession()->get(BootstrapMFAAuthenticator::SESSION_KEY . '.MemberID');
+
+        // First, let's see if we know the member
+        /** @var Member|null $member */
+        $member = Member::get()->byID($memberID);
+        if ($this->auditLogger) {
```
Comment (robbieaverill) [inline src/Handlers/BootstrapMFALoginHandler.php]: This property is undefined - you should define it and also add a public getter/setter for it

File: src/Handlers/BootstrapMFALoginHandler.php [hunk 2]
```diff
@@ -260,4 +272,37 @@ protected function cancelLogin(HTTPRequest $request)
...
+                case Logger::ALERT:
+                    $this->auditLogger->warn($message, $userInfo);
+                    break;
+                case Logger::EMERGENCY:
+                    $this->auditLogger->emergency($message, $userInfo);
+                    break;
+                default:
+                    // Default to a notice
+                    $this->auditLogger->notice($message, $userInfo);
+        }
```
Comment (robbieaverill) [inline src/Handlers/BootstrapMFALoginHandler.php]: Minor: phpcs violation

**Review by robbieaverill — CHANGES_REQUESTED** [review]
I've linked this to https://github.com/Firesphere/silverstripe-bootstrapmfa/issues/33 and left some minor comments on the proposed changes.

I'd note that the issue says "We should provide event hooks for https://github.com/silverstripe/silverstripe-auditor to pick up (and implement those listeners in the auditlog module)" - this is how the existing implementations of the auditor module work, and we should follow suit.

For example - where you've got `$this->log('Failure MFA login with ' . $authenticationMethod...` you'd instead use `$this->extend('onMFALoginFailure', $autheticationMethod)` or something, then implement the formatting logic into AuditHook, e.g. https://github.com/silverstripe/silverstripe-auditor/blob/846fdb30bf0c99f01d72ac21950eb997d3fc4ee0/code/AuditHook.php#L261-L280

**Comment by Firesphere** [comment]
> We should provide event hooks for https://github.com/silverstripe/silverstripe-auditor to pick up (and implement those listeners in the auditlog module)

Sure. I, in this case, "just" followed the auditor docs. I'll update accordingly

**Pull request closed without merging**